spring-cloud-kubernetes-commons/src/test/java/org/springframework/cloud/kubernetes/commons/profile/AbstractKubernetesProfileEnvironmentPostProcessorTest.java

```diff
@@ -0,0 +1,248 @@
+
+	private static final String PATH = "/some/path";
+
+	private MockedStatic<Paths> paths;
+
+	private MockedStatic<Files> files;
+
+	private final SpringApplication springApplication = Mockito.mock(SpringApplication.class);
+
+	private final ConfigurableApplicationContext context = new AnnotationConfigApplicationContext();
+
+	private static final AbstractKubernetesProfileEnvironmentPostProcessor POST_PROCESSOR_INSIDE = new AbstractKubernetesProfileEnvironmentPostProcessor() {
+		@Override
+		protected boolean isInsideKubernetes(Environment environment) {
+			return true;
+		}
+	};
+
+	private static final AbstractKubernetesProfileEnvironmentPostProcessor POST_PROCESSOR_OUTSIDE = new AbstractKubernetesProfileEnvironmentPostProcessor() {
+		@Override
+		protected boolean isInsideKubernetes(Environment environment) {
+			return false;
+		}
+	};
+
+	@Before
+	public void before() {
+		paths = Mockito.mockStatic(Paths.class);
+		files = Mockito.mockStatic(Files.class);
+	}
+
+	@After
+	public void after() {
+		paths.close();
+		files.close();
+	}
+
+	/**
+	 * <pre>
+	 * 1) "spring.cloud.kubernetes.enabled" is false; thus nothing happens
+	 * </pre>
+	 */
+	@Test
+	public void testKubernetesDisabled() {
+		TestPropertySourceUtils.addInlinedPropertiesToEnvironment(context, "spring.cloud.kubernetes.enabled=false");
+		POST_PROCESSOR_INSIDE.postProcessEnvironment(context.getEnvironment(), springApplication);
+
+		assertKubernetesProfileNotPresent();
+		assertKubernetesPropertySourceNotPresent();
+
+	}
+
+	/**
+	 * <pre>
+	 * 1) "spring.cloud.kubernetes.enabled" is true
+	 * 2) "spring.cloud.kubernetes.client.serviceAccountNamespacePath" is present, but does not resolve to an actual File
+	 * </pre>
+	 */
+	@Test
+	public void testKubernetesEnabledAndServiceAccountNamespacePathIsNotResolved() {
+		TestPropertySourceUtils.addInlinedPropertiesToEnvironment(context, "spring.cloud.kubernetes.enabled=true",
+				"spring.cloud.kubernetes.client.serviceAccountNamespacePath=" + PATH);
+		serviceAccountFileResolved(false, PATH);
+		POST_PROCESSOR_INSIDE.postProcessEnvironment(context.getEnvironment(), springApplication);
+
+		assertKubernetesProfilePresent();
+		assertKubernetesPropertySourceNotPresent();
+
+	}
+
+	/**
+	 * <pre>
+	 * 1) "spring.cloud.kubernetes.enabled" is true
+	 * 2) "spring.cloud.kubernetes.client.serviceAccountNamespacePath" is present and resolves to an actual File
+	 * </pre>
+	 */
+	@Test
+	public void testKubernetesEnabledAndServiceAccountNamespacePathIsResolved() {
+		TestPropertySourceUtils.addInlinedPropertiesToEnvironment(context, "spring.cloud.kubernetes.enabled=true",
+				"spring.cloud.kubernetes.client.serviceAccountNamespacePath=" + PATH);
+
+		Path path = serviceAccountFileResolved(true, PATH);
+		mockServiceAccountNamespace(path);
+
+		POST_PROCESSOR_INSIDE.postProcessEnvironment(context.getEnvironment(), springApplication);
+
+		assertKubernetesProfilePresent();
+		assertKubernetesPropertySourcePresent();
+
+	}
+
```

Comment:
I have tried to make the tests as readable as possible. 

 - So they have a minimal documentation that shows exactly the case this test tries to enforce
 - They all follow the same pattern, of first calling `TestPropertySourceUtils.addInlinedPropertiesToEnvironment`; thus setting in the environment the needed properties
 - Then they call the needed methods so that mocking would happen (`mockServiceAccountNamespace` for example)
 - Then they assert the presence of the needed invariants via `assertKubernetesProfilePresent` and `assertKubernetesPropertySourcePresent`

imho, this is clear to read and maintain.